addons/core/functions/common/fn_edenCreateAssets.sqf

```diff
@@ -0,0 +1,141 @@
+/* ----------------------------------------------------------------------------
+Function: bnb_es_core_fnc_edenCreateAssets
+
+Description:
+	Creates mission slots, modules and markers.
+
+Parameters:
+	0: _callsign - The platoon callsign that all sections will use <String>
+	1: _zeusCallsign - The callsign that HQ will use <String>
+	2: _numberOfSections - The number of sections to spawn <Number>
+
+Returns:
+	Nothing.
+
+Examples:
+	["Odin", "Valhalla", 3] call bnb_es_core_fnc_edenCreateAssets;
+
+Author:
+	Met
+---------------------------------------------------------------------------- */
+params [["_callsign", "Odin"],["_zeusCallsign", "Valhalla"],["_numberOfSections", 3]];
```

Comment:
If I were you I'd add the type checks too, so that ONLY strings are accepted. So you'd modify each element there to have a third item `["_callsign", "Odin", [""]]`.

Also, is there no way to have the params function handle the empty string issue? Maybe test to see if you can stick a count limit even if it is a string and not an array?

Reply:
Added type checks as suggested, thanks. 

Unfortunately, expectedArrayCount doesn't appear to work on strings, but I could instead not allow the user to press "Ok" until they input a value in the required fields.

Reply:
Nah, if the counter doesn't work for strings then I suggest leave as is. It's better to make sure these checks are handled at entry to a function, because there's no guarantee that the interface is the ONLY place from where this function could be called. E.g. Debugging or future implementations that would like to use this functionality.

As for making sure there's something in the field in the interface itself, it's always a good idea to provide the user with as much feedback as possible. So just make sure there's a note on the interface stating that Odin is default and then all good